File: spring-data-requery/src/test/java/org/springframework/data/requery/domain/model/AbstractAddress.java
package org.springframework.data.requery.domain.model;

import io.requery.*;
import lombok.Getter;
import lombok.Setter;

/**
 * org.springframework.data.requery.domain.model.AbstractBasicLocation
 *
 * @author debop
 * @since 18. 6. 4
 */
@Getter
@Setter
@Entity(copyable = true)
public abstract class AbstractAddress extends Coordinate {

    @Key
    @Generated
    protected int id;

    protected String line1;
    protected String line2;

    protected String state;

    @Column(length = 5)
    protected String zip;

    @Column(length = 2)
    protected String countryCode;

    protected String city;

    @OneToOne(mappedBy = "address", cascade = { CascadeAction.DELETE, CascadeAction.SAVE })
    protected AbstractPerson person;

    @Convert(AddressTypeConverter.class)
    protected AddressType type;

    @Override
    public String toString() {
        return "Don't override me";
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/domain/sample/AbstractAddress.kt
package org.springframework.data.requery.kotlin.domain.sample

import io.requery.Column
import io.requery.Embedded
import io.requery.Transient
import org.springframework.data.requery.kotlin.domain.AbstractComponent
import org.springframework.data.requery.kotlin.domain.ToStringBuilder

@Embedded
abstract class AbstractAddress: AbstractComponent() {

    @get:Column(length = 128)
    abstract var country: String?

    @get:Column(length = 64)
    abstract var city: String?

    @get:Column(length = 1024)
    abstract var streetName: String?

    @get:Column(length = 1024)
    abstract var streetNo: String?

    @Transient
    override fun buildStringHelper(): ToStringBuilder {
        return super.buildStringHelper()
            .add("country", country)
            .add("city", city)
            .add("streetName", streetName)
            .add("streetNo", streetNo)
    }
}